library/core/src/slice/iter.rs

```diff
@@ -2067,15 +2051,11 @@ impl<'a, T> Iterator for ChunksExactMut<'a, T> {
 
     #[inline]
     fn next(&mut self) -> Option<&'a mut [T]> {
-        if self.v.len() < self.chunk_size {
-            None
-        } else {
-            // SAFETY: self.chunk_size is inbounds because we compared above against self.v.len()
-            let (head, tail) = unsafe { self.v.split_at_mut(self.chunk_size) };
-            self.v = tail;
-            // SAFETY: Nothing else points to or will point to the contents of this slice.
-            Some(unsafe { &mut *head })
-        }
+        // SAFETY: we have `&mut self`, so are allowed to temporarily materialize a mut slice
```

Comment:
I'd add that `next` must not be called if `__get_unchecked` was called, so invalidating previously returned items does not conflict with `TrustedRandomAccess`.

Reply:
While that is true, I don't see how this particular `next` method is special in that regard. So would you then like all `next` methods of iterators that impl `TrustedRandomAccess` to get such a comment?